src/DotNetWorker.Core/Converters/Converter/DefaultInputConverterProvider.cs

```diff
@@ -50,12 +50,30 @@ public IEnumerable<IInputConverter> RegisteredInputConverters
                 {
                     yield return _converterCache.GetOrAdd(converterType.AssemblyQualifiedName!, (key) =>
                     {
-                        return (IInputConverter)ActivatorUtilities.CreateInstance(_serviceProvider, converterType);
+                        return GetOrCreateConverterInstance(converterType);
                     });
                 }
             }
         }
 
+        /// <summary>
+        /// Gets an instance of the converter for the type requested.
+        /// </summary>
+        /// <param name="converterType">The type for which we are requesting an IInputConverter instance.</param>
+        /// <exception cref="InvalidOperationException">Throws when the converterType param is null.</exception>
+        /// <returns>IConverter instance of the requested type.</returns>
+        public IInputConverter GetOrCreateConverterInstance(Type converterType)
+        {
+            if (converterType is null)
+            {
+                throw new InvalidOperationException($"Could not create an instance of {(nameof(converterType))}.");
+            }
+
+            EnsureTypeCanBeAssigned(converterType);
+
+            return (IInputConverter)ActivatorUtilities.CreateInstance(_serviceProvider, converterType);
```

Comment:
I did not completely follow the change being recommended here.

Reply:
Okay I see what you are saying. I have created this method `GetOrCreateConverterInstance(Type converterType)` as an overload of another method in this class `GetOrCreateConverterInstance(string converterTypeName)` (just below this method) and trying to do the exact same thing in this method.

Reply:
Shouldn't the other overload be refactored to call this one then - after resolving its type name to a Type, call this overload?

Reply:
I made the change so that string overload will call the Type overload and I agree with the change to remove string overload. But it's kind of making me change 4-5 files. Let me know if I should update this PR or refactor it separately (I can start another PR with only those changes quickly.)